File: server/Optika.API/Optika.API/DTOs/OrderCreateDto.cs
using System.ComponentModel.DataAnnotations;

namespace Optika.API.DTOs
{
    public class OrderCreateDto
    {
        [Required]
        public int UserId { get; set; }

        [Required]
        public ICollection<OrderItemCreateDto> Items { get; set; } = new List<OrderItemCreateDto>();

        [Required]
        [Range(0.01, double.MaxValue, ErrorMessage = "TotalPrice must be greater than 0.")]
        public decimal TotalPrice { get; set; }

        [Required]
        [MaxLength(50)]
        public string Status { get; set; } = "Pending";
    }
}

File: client/lib/models/order_create_dto.dart
import 'order_item_create_dto.dart';

class OrderCreateDto {
  final int userId;
  final List<OrderItemCreateDto> items;
  final double totalPrice;
  final String status;

  OrderCreateDto({
    required this.userId,
    required this.items,
    required this.totalPrice,
    this.status = 'Pending',
  });

  Map<String, dynamic> toJson() => {
    'userId': userId,
    'items': items.map((i) => i.toJson()).toList(),
    'totalPrice': totalPrice,
    'status': status,
  };
}
